### ISSUE TITLE
New-HPOVApplianceCsr - The "type" field content is invalid

### ISSUE THREAD

**Issue by salcinad** [HewlettPackard/POSH-HPOneView#431]
Trying to generate a appliance CSR (HPE Synergy Composer2) but facing issue with example provided.

I also used real data and not just these provided in example, same error message about type filed of the JSON.

```
$CSR = @{
    Country = "US";
    State = "California";
    City = "Palo Alto";
    Organization = "Hewlett-Packard";
    CommonName = "hpov.example.com";
    AlternativeName = "hpov.example.com,192.168.1.1,hpov"
}    
    
$request = New-HPOVApplianceCsr @CSR
New-HPOVApplianceCsr : The "type" field content, "CertificateSigningRequest", included in this request is invalid. Correct the content in the 'type' field of the JSON and try the request again. 
At line:11 char:12
+ $request = New-HPOVApplianceCsr @CSR
+            ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Send-HPOVRequest:String) [New-HPOVApplianceCsr], InvalidOperationException
    + FullyQualifiedErrorId : INVALID_JSON_TYPE,New-HPOVApplianceCsr 
```

HPE Synergy Composer2 version and POSH HPOneView module.
```
Get-HPOVVersion -ApplianceVer

syn01.somedomain.com           LibraryVersion Path                                                                  
-------------------------           -------------- ----                                                                  
ApplianceVersion: 5.00.00.400525.00 5.0.2226.2609  C:\Program Files\WindowsPowerShell\Modules\HPOneView.500\5.0.2226.2609

```

Also using just a whole command without $CSR give us same problem.

When using Get-HPOVCommandTrace we see following messages (croped to just VERBOSE part), if needed we can provide per mail the completed trace log.

```
.....
VERBOSE: [SEND-HPOVREQUEST] Net.WebException Error caught
VERBOSE: [SEND-HPOVREQUEST] Exception Object: 

PSMessageDetails      : 
Exception             : System.Net.WebException: The remote server returned an error: (400) Bad Request.
                           at System.Net.HttpWebRequest.GetResponse()
                           at CallSite.Target(Closure , CallSite , WebRequest )
TargetObject          : 
CategoryInfo          : NotSpecified: (:) [], WebException
FullyQualifiedErrorId : WebException
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Send-HPOVRequest<Process>, C:\Program Files\WindowsPowerShell\Modules\HPOneView.500\5.0.2226.2609\HPOneView.500.psm1: line 5533
                        at New-HPOVApplianceCsr<Process>, C:\Program Files\WindowsPowerShell\Modules\HPOneView.500\5.0.2226.2609\HPOneView.500.psm1: line 16781
                        at <ScriptBlock>, <No file>: line 2
                        at Get-HPOVCommandTrace<Process>, C:\Program Files\WindowsPowerShell\Modules\HPOneView.500\5.0.2226.2609\HPOneView.500.psm1: line 9443
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}
VERBOSE: [SEND-HPOVREQUEST] Exception Message: The remote server returned an error: (400) Bad Request.
VERBOSE: [SEND-HPOVREQUEST] InnerException FullyQualifiedErrorId:
VERBOSE: [SEND-HPOVREQUEST] InnerException Message:
VERBOSE: [SEND-HPOVREQUEST] Exception
VERBOSE: [SEND-HPOVREQUEST] Getting Error Response
VERBOSE: [SEND-HPOVREQUEST] ERROR RESPONSE: {"errorCode":"INVALID_JSON_TYPE","message":"Invalid JSON type.","details":"The \"type\" field content, \"CertificateSigningRequest\", included in this request is invalid.","recommendedActions":["Correct the content in the 'type' field of the JSON and try the request again."],"errorSource":null,"nestedErrors":[],"data":{}}
VERBOSE: [SEND-HPOVREQUEST] Response Status: HTTP 400 [Bad Request]
VERBOSE: [SEND-HPOVREQUEST] Response Header: Pragma = Mon, 25 Nov 2019 11:48:46 GMT
VERBOSE: [SEND-HPOVREQUEST] Response Header: X-XSS-Protection = 0
VERBOSE: [SEND-HPOVREQUEST] Response Header: X-Frame-Options = Apache
VERBOSE: [SEND-HPOVREQUEST] Response Header: X-Content-Type-Options =
VERBOSE: [SEND-HPOVREQUEST] Response Header: Connection =
VERBOSE: [SEND-HPOVREQUEST] Response Header: Transfer-Encoding =
VERBOSE: [SEND-HPOVREQUEST] Response Header: Cache-Control =
VERBOSE: [SEND-HPOVREQUEST] Response Header: Content-Type =
VERBOSE: [SEND-HPOVREQUEST] Response Header: Date =
VERBOSE: [SEND-HPOVREQUEST] Response Header: Expires =
VERBOSE: [SEND-HPOVREQUEST] Response Header: Server =
VERBOSE: [SEND-HPOVREQUEST] HTTP 400 error caught.
VERBOSE: [NEW-ERRORRECORD] Building ErrorRecord object
VERBOSE: [SEND-HPOVREQUEST] Cleaning up HttpWebRequest
.....
```

**Comment by ChrisLynchHPE** [comment]
This is a bug with the API request the Cmdlet creates.  It will be fixed in the next update.

**Comment by radekzima** [comment]
Hello Chris, please change the type in the json to "CertificateSigningRequestV2", the New-HPOVAapplianceCsr fails otherwise and we have to use the plain REST ... thanks a million, Radek

**Comment by ChrisLynchHPE** [comment]
@radekzima , yes, that is the fix I have already done.  The library update has not been released yet.  I am looking to do that in the next few days.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 5.0.2295.3359](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v5.0.2295.3359) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.500/5.0.2295.3359).

Please verify before this issue is closed.

**Comment by salcinad** [comment]
I just verified this and I can confirm that this is working for me now


```
Get-HPOVVersion -ApplianceVer

syn01.somedomain.com          LibraryVersion Path                                                                  
-------------------------           -------------- ----                                                                  
ApplianceVersion: 5.00.01.410269.00 5.0.2295.3359  C:\Program Files\WindowsPowerShell\Modules\HPOneView.500\5.0.2295.3359


$request
type                : CertificateDataV2
uri                 : /rest/certificates
category            : appliance
eTag                : 2020-01-15T06:44:17.465Z
created             : 2020-01-15T06:44:17.465Z
modified            : 2020-01-15T06:44:17.465Z
base64Data          : -----BEGIN CERTIFICATE REQUEST-----
.............
.............(deleted)
                      -----END CERTIFICATE REQUEST-----
description         : 
state               : 
name                : 
status              : 
ApplianceConnection : syn01.somedomain.com
```
